packages/core/utils/src/modules-sdk/medusa-service.ts

```diff
@@ -243,7 +243,7 @@ export function MedusaService<
           data = [],
           sharedContext: Context = {}
         ): Promise<T | T[]> {
-          const serviceData = Array.isArray(data) ? data : [data]
+          const serviceData = data
```

Comment:
@adrien2p, should we add a few tests for this before we merge?

Reply:
Yea definitely 💯 forgot to mention it. Also, removing the service data const as it does not do anything anymore

Reply:
cc @dwene